code/game/gamemodes/objective.dm

```diff
@@ -5,6 +5,7 @@ datum/objective
 	var/target_amount = 0				//If they are focused on a particular number. Steal objectives have their own counter.
 	var/completed = 0					//currently only used for custom objectives.
 	var/dangerrating = 0				//How hard the objective is, essentially. Used for dishing out objectives and checking overall victory.
+	var/martyr_compatible = 1			//If the objective is compatible with martyr objective, i.e. if you can still do it while dead.

```

Comment:
Since Martyr is set to only trigger instead of Hijack, it's better to set flags as exclusionary, rather than inclusionary. Better to say "You can martyr if it's not steal/debrain/survive", rather than "You cannot martyr unless assassinate"